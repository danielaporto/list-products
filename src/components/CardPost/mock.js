export const cardPostMock = {
  posts: {
    userId: 10,
    id: 1,
    title: 'Title 1',
    body: 'Body 1',
    cover: 'img/img.webp',
  },
};
