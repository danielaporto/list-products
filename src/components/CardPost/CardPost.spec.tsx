import React from 'react';
import { render, screen } from '@testing-library/react';
import { CardPost } from '.';
import { cardPostMock } from './mock';

describe('<CardPost />', () => {
  it('should render <CardPost />', () => {
    render(<CardPost {...cardPostMock} />);
    expect(screen.getByAltText(/title 1/i)).toHaveAttribute('src', 'img/img.webp');
    expect(screen.getByRole('heading', { name: 'Title 1' })).toBeInTheDocument();
    expect(screen.getByText('Body 1')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<CardPost {...cardPostMock} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
