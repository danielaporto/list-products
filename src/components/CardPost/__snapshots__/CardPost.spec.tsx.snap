// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardPost /> should match snapshot 1`] = `
<div
  class="sc-bcXHqe dmXgPZ CardPost"
>
  <img
    alt="Title 1"
    class="sc-gswNZR fYcGdc"
    src="img/img.webp"
  />
  <h1
    class="sc-dkrFOg"
    style="overflow: hidden; display: -webkit-box;"
  >
    Title 1
  </h1>
  <p
    class="sc-hLBbgP"
    style="overflow: hidden; display: -webkit-box;"
  >
    Body 1
  </p>
</div>
`;

exports[`<CardPost /> should render <CardPost /> 1`] = `
<div
  class="sc-bcXHqe dmXgPZ CardPost"
>
  <img
    alt="Title 1"
    class="sc-gswNZR fYcGdc"
    src="img/img.webp"
  />
  <h1
    class="sc-dkrFOg"
    style="overflow: hidden; display: -webkit-box;"
  >
    Title 1
  </h1>
  <p
    class="sc-hLBbgP"
    style="overflow: hidden; display: -webkit-box;"
  >
    Body 1
  </p>
</div>
`;
